pkg/app/server/grpcapi/api.go

```diff
@@ -381,6 +382,149 @@ func (a *API) GetDeployment(ctx context.Context, req *apiservice.GetDeploymentRe
 	}, nil
 }
 
+func (a *API) ListDeployments(ctx context.Context, req *apiservice.ListDeploymentsRequest) (*apiservice.ListDeploymentsResponse, error) {
+	key, err := requireAPIKey(ctx, model.APIKey_READ_ONLY, a.logger)
+	if err != nil {
+		return nil, err
+	}
+
+	orders := []datastore.Order{
+		{
+			Field:     "UpdatedAt",
+			Direction: datastore.Desc,
+		},
+		{
+			Field:     "Id",
+			Direction: datastore.Asc,
+		},
+	}
+	filters := []datastore.ListFilter{
+		{
+			Field:    "ProjectId",
+			Operator: datastore.OperatorEqual,
+			Value:    key.ProjectId,
+		},
+	}
+	// Allowing multiple so that it can do In Query later.
+	// Currently only the first value is used.
+	if len(req.Statuses) > 0 {
+		statuses := []model.DeploymentStatus{}
+		for _, s := range req.Statuses {
+			if s != "" {
+				depstatus, ok := model.DeploymentStatus_value[s]
+				if !ok {
+					return nil, status.Errorf(codes.InvalidArgument, "%s is invalid deployment status", s)
+				}
+				statuses = append(statuses, model.DeploymentStatus(depstatus))
+			}
+		}
+
+		filters = append(filters, datastore.ListFilter{
+			Field:    "Status",
+			Operator: datastore.OperatorEqual,
+			Value:    statuses[0],
+		})
+	}
+	if len(req.Kinds) > 0 {
+		kinds := []model.ApplicationKind{}
+		for _, k := range req.Kinds {
+			if k != "" {
+				kind, ok := model.ApplicationKind_value[k]
+				if !ok {
+					return nil, status.Errorf(codes.InvalidArgument, "%s is invalid application kind", k)
+				}
+				kinds = append(kinds, model.ApplicationKind(kind))
+			}
+		}
+
+		filters = append(filters, datastore.ListFilter{
+			Field:    "Kind",
+			Operator: datastore.OperatorEqual,
+			Value:    kinds[0],
+		})
+	}
+	if len(req.ApplicationIds) > 0 {
+		filters = append(filters, datastore.ListFilter{
+			Field:    "ApplicationId",
+			Operator: datastore.OperatorEqual,
+			Value:    req.ApplicationIds[0],
+		})
+	}
+	if req.ApplicationName != "" {
+		filters = append(filters, datastore.ListFilter{
+			Field:    "ApplicationName",
+			Operator: datastore.OperatorEqual,
+			Value:    req.ApplicationName,
+		})
+	}
+
+	// The default value of limit is 30.
+	limit := 30
```

Comment:
Please see this doc, you'll get to know how to set the default value.
https://pkg.go.dev/github.com/spf13/pflag#Int32Var